src/Adapter/MSTest.TestAdapter/Helpers/ReflectHelper.cs

```diff
@@ -206,90 +197,121 @@ public bool HasAttributeDerivedFrom<TAttribute>(MemberInfo memberInfo, bool inhe
 #endif
     public override object InitializeLifetimeService() => null!;
 
-    internal static TAttribute[]? GetAttributes<TAttribute>(MethodBase methodBase, bool inherit)
-        where TAttribute : Attribute
-    {
-        TAttribute[]? attributeArray = GetCustomAttributes<TAttribute>(methodBase, inherit);
-        return attributeArray == null || attributeArray.Length == 0
-            ? null
-            : attributeArray;
-    }
-
     /// <summary>
-    /// Match return type of method.
+    /// Gets single attribute, DON'T use this together with attribute that does not allow multiple, to avoid eating the cost of checking for more when there cannot be more.
```

Comment:
This method is unused, and can be removed, but keeping it in case we find I should have used it somewhere. But likely not, because it is only useful for attributes that allow multliple, and we already grab those as collection and do custom validation on them.